cadl-extension/src/code-model-builder.ts

```diff
@@ -132,11 +134,20 @@ export class CodeModelBuilder {
 
   private specialHeaderNames = new Set(["repeatability-request-id", "repeatability-first-sent"]);
 
-  public constructor(program1: Program, options: EmitterOptions) {
-    this.options = options;
+  
+  public constructor(program1: Program, context: EmitContext<EmitterOptions>) {
+    this.options = context.options;
     this.program = program1;
-    this.dpgContext = { program: this.program, generateProtocolMethods: true, generateConvenienceMethods: true };
 
+    const dpgEmitterOptions = {
+      "generate-protocol-methods": true,
+      "generate-convenience-methods": this.options["dev-options"]?.["generate-convenience-apis"] ?? true
+    };
```

Comment:
I don't get it. I thought the point is that cadl-dpg can use their code to get the configure, instead of we set the configure https://github.com/Azure/cadl-azure/blob/main/packages/cadl-dpg/src/decorators.ts#L245-L246